src/connection.ts

```diff
@@ -64,30 +64,13 @@ export function newConnection(input: any, output: any, trace: TraceOptions): ICo
 	}
 
 	const connection = createConnection(reader, writer);
+
+	// Remove vscode-languageserver's stream listeners that kill the process if the stream is closed
 	input.removeAllListeners('end');
 	input.removeAllListeners('close');
 	output.removeAllListeners('end');
 	output.removeAllListeners('close');
 
-	let closed = false;
-	function close() {
-		if (!closed) {
-			if ((<fs_.ReadStream>input).close) {
-				(<fs_.ReadStream>input).close();
-			}
-			if ((<fs_.WriteStream>output).close) {
-				(<fs_.WriteStream>output).close();
-			}
-			closed = true;
-		}
-	}
-
-	// We attach one notification handler on `exit` here to handle the
-	// teardown of the connection.  If other handlers want to do
-	// something on connection destruction, they should register a
-	// handler on `shutdown`.
-	connection.onNotification(rt.ExitRequest.type, close);
```

Comment:
this is now handled in `server.ts` so we close the socket correctly (it's `socket.end()`, not `socket.close()`) and we don't override the exit handler (as vscode-languageserver/json-rpc only allows a single handler per method). Previously because of this code here the STDIO version would not exit on an exit notification, which the child processes need to do.